### PR TITLE
Make the classifier's subject viewer sticky

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
lib-classifier

## Linked Issue and/or Talk Post
Toward: https://github.com/zooniverse/front-end-monorepo/issues/6063

## Describe your changes
- This change is intended to improve volunteer's classifying experience when a workflow has a "tall" task area and landscape oriented subject viewer. For example, when a volunteer clicks "Done" on a survey task, we want them to keep the subject in view rather than forced to scroll up and down the page to complete each classification.
- Added sticky css styling to the ViewerGrid component for NoMaxWidth and MaxWidth layout. MaxWidth layout is the default layout and NoMaxWidth layout is specific to transcription projects.
- The CenteredLayout's task area wasn't being sticky (e.g Daily Minor Planet) especially for separate frames view, so I fixed that. CenteredLayout only applies when a project team configures their workflow to "limit subject height".

## How to Review

A few types of projects to review are below. The full list of FEM projects is [here](https://docs.google.com/spreadsheets/d/1X9345vQ_xNX6BJf9_xwVCQCjxjbWWErGmV7Xe8WpHZo/edit#gid=0).
- [PH TESS](https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess?env=production&demo=true)
- [Daily Minor Planet](https://local.zooniverse.org:3000/projects/fulsdavid/the-daily-minor-planet/classify/workflow/22354?env=production&demo=true)
- [PRINT](https://local.zooniverse.org:3000/projects/printmigrationnetwork/print?env=production&demo=true)
- [Wildwatch Burrowing Owl](https://local.zooniverse.org:3000/projects/sandiegozooglobal/wildwatch-burrowing-owl?env=production&demo=true)
- [Squirrel Mapper](https://local.zooniverse.org:3000/projects/bcosentino/squirrelmapper?env=production&demo=true)

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/Layout/helpers/getLayout/getLayout.js [hunk 1]
```diff
@@ -4,7 +4,6 @@ export default function getLayout(layout) {
   if (layouts[layout]) {
     return layouts[layout]
   } else {
-    console.warn(`Couldn't find a layout for '${layout}', falling back to default`)
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/components/Classifier/components/Layout/helpers/getLayout/getLayout.js:7]: Opinionated statement, but this warning was just cluttering the console during unit tests.

**Comment by seanmiller26** [comment]
This is a BIG step in the right direction. Super useful to keep the subject in view at all times. I tried every preview link and they all work great. OK by me to move forward with implementing this.

**Comment by seanmiller26** [comment]
I have 2 small questions about page positioning and scrolling which I chatted about briefly with Delilah. 

1. When submitting a classification, you will not be brought back up to the top of the page and the new subject sometimes loads out of view.
https://github.com/zooniverse/front-end-monorepo/assets/89043489/45ffce49-48b4-4a26-b390-2a41a84b0a16

2. When you are in a survey selection modal and you choose 'Not this,' you are brought back to the very top of the page. It can be a pretty jarring interaction.

I'd love to unify the behavior here. Either they both bring you back to the top of the page (preferably just above the subject, no need to see the header and project header), or you will retain your page position for both.

**Comment by goplayoutside3** [comment]
1. I think the video example here illustrates behavior that is purposely in the volunteer's control. If a volunteer clicks Done while the Done button is in the vertical middle of their screen, then it's not a problem for the Done button to remain in the vertical middle of their screen.

2. However, this video is something I'll look into because it's also jarring on production, so seems to be caused by something else regardless of a sticky subject viewer. Will report back!

**Comment by goplayoutside3** [comment]
The jarring scroll-to-top behavior with the survey task specifically is because the Chooser component (all the animal options), and the Choice component (form that collects final answer) are drastically different heights. Only one renders at a time, and usually the Choice component is much taller than the Chooser because it includes example images, paragraphs, select buttons, etc. If a volunteer selects an animal to identify, they see the Choice component, scroll down so the "Identify" button is in the middle of their screen vertically, click "Identify", but then the Choice component is replaced with the shorter Chooser component. This throws off window positioning regardless of whether the subject area is sticky. One way to solve this is to make the survey tasks's Choice component scrollable in an isolated manner, rather than scrolling the entire page. 

However, there's a similar problem for projects that have a super tall task area like: `md68135/notes-from-nature-labs` which has a combo task of 4 single answer questions. A volunteer must scroll down the page to answer all four questions and click Next or Done which causes the window to scroll almost all the way to the top. 

To summarize, the cause of the window jumping back up to the top when a volunteer clicks buttons in the task area is because it changes height while the subject image height remains the same, and the browser takes a best guess about position. In a perfect world, all components of the classifier would be designed to fit inside a volunteer's screen height for an immersive experience. Making the subject viewer sticky is a step toward that and would still like this PR reviewed as is.

I'm hesitant to manually manipulate users' scroll position, but something we could explore is a function triggered by Done, Next, Back, and the survey task Identify buttons that says "if the top edge of the classifier is above a user's viewport, scroll their screen so the top edge of the classifier is at the top edge of their viewport". In a separate PR though.

**Review by mcbouslog — APPROVED** [review]
This looks good! Tested on noted projects without issue.

**Comment by mcbouslog** [comment]
Are we concerned with mobile browser behavior?
The task area covers the subject on mobile.

Uploading TESSmobile.mp4…

**Comment by goplayoutside3** [comment]
Hmmm I think I see what you mean on mobile. The video in your comment didn't upload, but I'll remove all sticky behavior for mobile-sized viewports before merging this PR!

**Pull request merged**